setup.py

```diff
@@ -1,12 +1,20 @@
 #!/usr/bin/env python
 
-import os
-import sys
 import io
+import sys
+import warnings
 
 from setuptools import setup, find_packages
 
 
+if sys.version_info[:2] <= (2, 6) or ((3, 0) <= sys.version_info[:2] <= (3, 3)):
+    python_ver = '.'.join(str(x) for x in sys.version_info[:3])
+
+    warnings.warn('You are using Python {0}, which will no longer be supported in '
+                  'JMESPath 1.0'.format(python_ver),
+                  DeprecationWarning)
```

Comment:
I would prefer we start with a `PendingDeprecationWarning`.  We can then document the deprecation process in docs before starting to deprecate.

Reply:
Updated!